packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -367,16 +367,17 @@ export class DataStores implements IDisposable {
         return summaryBuilder.getSummaryTree();
     }
 
-    public createSummary(): ISummaryTreeWithStats {
+    public async createSummary(): Promise<ISummaryTreeWithStats> {
```

Comment:
Same comment as above: It's possible to ensure (by code inspection) that this function returns state of container at the time when createSummary() call started. But we will not be able to sustain that over time, even with UTs. I believe if we go that route, we have to commit (and implement quickly as a follow up) same flow you are implementing for DDS here, i.e., for whole container, with synchronous "capture revision" and async "generate snapshot out of it.